src/store/token.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+export async function getToken(token) {
+	const tokenValue = await ((typeof (token) === 'function')
+		? token()
+		: token);
+
+	return await new Token(tokenValue);
+}
+
+export function compareTokens(a, b) {
+	return a.toString() === b.toString();
+}
+
+export const TOKEN_COOKIE = -1;
+export const TOKEN_COOKIE_CACHE_KEY = 'cookie';
+
+class Token {
+	constructor(token) {
+		this._cacheKey = this._praseCacheKey(token);
+		this._value = token;
+	}
+
+	get value() {
+		return this._value;
+	}
+
+	toString() {
+		return this._cacheKey;
+	}
+
+	isResolved() {
```

Comment:
What does a resolved token mean?

Reply:
That it has been processed with no errors.

Reply:
Is this token a valid instance of `Token`.